internal/controller/controllers/agent_controller.go

```diff
@@ -393,30 +428,36 @@ func (r *AgentReconciler) deregisterHostIfNeeded(ctx context.Context, log logrus
 	return buildReply(nil)
 }
 
-func (r *AgentReconciler) isDay2NonePlatformHostRebooting(ctx context.Context, agent *aiv1beta1.Agent, h *models.Host) (bool, error) {
-	if swag.StringValue(h.Status) == models.HostStatusAddedToExistingCluster &&
-		h.Progress.CurrentStage == models.HostStageDone {
-		if agent.Status.Progress.CurrentStage == models.HostStageDone {
-			return false, nil
-		} else {
-			isNone, err := isAgentInNonePlatformCluster(ctx, r.Client, agent)
-			if err != nil {
-				return false, err
-			}
-			return isNone, nil
+func (r *AgentReconciler) isRebootingHost(ctx context.Context, agent *aiv1beta1.Agent, h *models.Host) (bool, error) {
```

Comment:
For day2 host, it moves to Done when it starts rebooting. Here https://github.com/openshift/assisted-service/blob/b31fbf34d02bf54e278276af53cea6a3e484baa6/internal/host/host.go#L571
Better call it isDay2HostRebooting because it is irrelevant to day1

Reply:
And please return the check for day2 
**if swag.StringValue(h.Status) == models.HostStatusAddedToExistingCluster**